### PR TITLE
Issue #15079: Add new input file for sealed classes in InterfaceMembe…

### LINKED ISSUE #15079: Add Check Support for Java 17 Sealed Classes: InterfaceMemberImpliedModifier
child of https://github.com/checkstyle/checkstyle/issues/14969
I have read check documentation: https://checkstyle.org/checks/modifier/interfacememberimpliedmodifier.html#InterfaceMemberImpliedModifier

---

```
PS D:\CS\test> cat config.xml                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="InterfaceMemberImpliedModifier">

        </module>
    </module>
</module>
PS D:\CS\test> cat src/Test.java                                                
sealed interface I permits A {
    String s = null; // 3 violations
    void m(); // 2 violations
    sealed interface J permits A {  // 2 violations
        void m();  // 2 violations
    }
}
final class A implements I, I.J {
    @Override
    public void m() {
    }
}
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:2:5: Implied modifier 'final' should be explicit. [InterfaceMemberImpliedModifier]
[ERROR] D:\CS\test\src\Test.java:2:5: Implied modifier 'public' should be explicit. [InterfaceMemberImpliedModifier]
[ERROR] D:\CS\test\src\Test.java:2:5: Implied modifier 'static' should be explicit. [InterfaceMemberImpliedModifier]
[ERROR] D:\CS\test\src\Test.java:3:5: Implied modifier 'abstract' should be explicit. [InterfaceMemberImpliedModifier]
[ERROR] D:\CS\test\src\Test.java:3:5: Implied modifier 'public' should be explicit. [InterfaceMemberImpliedModifier]
[ERROR] D:\CS\test\src\Test.java:4:5: Implied modifier 'public' should be explicit. [InterfaceMemberImpliedModifier]
[ERROR] D:\CS\test\src\Test.java:4:5: Implied modifier 'static' should be explicit. [InterfaceMemberImpliedModifier]
[ERROR] D:\CS\test\src\Test.java:5:9: Implied modifier 'abstract' should be explicit. [InterfaceMemberImpliedModifier]
[ERROR] D:\CS\test\src\Test.java:5:9: Implied modifier 'public' should be explicit. [InterfaceMemberImpliedModifier]
Audit done.
Checkstyle ends with 9 errors.
PS D:\CS\test> 

```

---

the check works as intended in sealed interfaces.

### PR DESCRIPTION
**Pull request by mahfouz72**
closes #15079

### CODE REVIEW

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+interface InputInterfaceMemberImpliedModifierSealedClassesCorrected {
+    final public static String s = null;
+    abstract public void m();
+    public  static  sealed interface J permits A {
+        abstract public void m();
+    }
+    public static sealed class B permits D { }
+   public static non-sealed class D extends B { }
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java:46]: Indentation is off.

**Review by rnveach — CHANGES_REQUESTED** [review]

**Review by nrmancuso — APPROVED** [review]
I am good after https://github.com/checkstyle/checkstyle/pull/15312/files#r1683217722

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java:46]: done

Comment (rnveach) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java:46]: @mahfouz72 Not done. Indentation is worse now.

This seems to be a constant issue in PRs. You may want to keep a closer eye on this.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*
+InterfaceMemberImpliedModifier
+
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java:3]: I just noticed this now when reviewing another PR. How come this top portion doesn't mention the default properties?

I guess our tool doesn't ensure all properties are listed. We are missing double validation here.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java]: I hope it is done now :)

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java (hunk 2 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java:10]: Yeah done, I add the default properties

> I guess our tool doesn't ensure all properties are listed.  

yes

**Review by rnveach — APPROVED** [review]

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java (hunk 2 above)
Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/InputInterfaceMemberImpliedModifierSealedClasses.java:10]: Better to reference them, even it is not enforced

**Comment by mahfouz72** [comment]
This needs to be restarted circleci has stopped

**Comment by mahfouz72** [comment]
3 failures here are unrelated checks are stopped with no output

**Pull request merged**